src/django_components/dependencies.py

```diff
@@ -771,30 +789,28 @@ class ComponentDependencyMiddleware:
     """
 
     def __init__(self, get_response: "Callable[[HttpRequest], HttpResponse]") -> None:
-        self.get_response = get_response
+        self._get_response = get_response
 
         # NOTE: Required to work with async
-        if iscoroutinefunction(self.get_response):
+        if iscoroutinefunction(self._get_response):
             markcoroutinefunction(self)
 
     def __call__(self, request: HttpRequest) -> HttpResponseBase:
         if iscoroutinefunction(self):
             return self.__acall__(request)
 
-        response = self.get_response(request)
-        response = self.process_response(response)
+        response = self._get_response(request)
+        response = self._process_response(response)
         return response
 
     # NOTE: Required to work with async
     async def __acall__(self, request: HttpRequest) -> HttpResponseBase:
-        response = await self.get_response(request)
-        response = self.process_response(response)
+        response = await self._get_response(request)
+        response = self._process_response(response)
         return response
 
-    def process_response(self, response: HttpResponse) -> HttpResponse:
-        if not isinstance(response, StreamingHttpResponse) and response.get("Content-Type", "").startswith(
-            "text/html"
-        ):
+    def _process_response(self, response: HttpResponse) -> HttpResponse:
```

Comment:
Not related to the bug. Prefixed this method with underscore, so it does not appear in the generated documentation.